xmppserver/src/main/java/org/jivesoftware/util/ListPager.java

```diff
@@ -45,10 +45,21 @@
 
     private static final String REQUEST_PARAMETER_KEY_PAGE_SIZE = "listPagerPageSize";
     private static final String REQUEST_PARAMETER_KEY_CURRENT_PAGE = "listPagerCurrentPage";
+    private static final String REQUEST_PARAMETER_KEY_SORT_ORDER = "sortOrder";
+    private static final String REQUEST_PARAMETER_KEY_SORT_COLUMN_NUMBER = "sortColumnNumber";
     private static final String PAGINATION_FORM_ID = "paginationForm";
     private static final int DEFAULT_PAGE_SIZE = 25;
     private static final int[] PAGE_SIZES = {25, 50, 100, 1000};
 
+    /**
+     * Descending sort (ie 3, 2, 1...).
+     */
+    public static final int DESCENDING = 0;
```

Comment:
Perhaps an enum? Or given it only takes two values, would an Boolean suffice?

`private final Boolean sortAscending =` True, False or null (null implying don't care)?

Reply:
(and if you don't like "tri state" Booleans, perhaps an Enum to avoid having to catch the ~4 billion invalid values for an integer?)

Reply:
I agree that this isn't the most elegant of solutions. I used this, as this is also how it is used in `SessionResultFilter`. The numeric value is what is being passed as a query parameter in the JSPs, which is convenient.

I am liking the clarity that using words like 'DESCENDING' or 'ASCENDING' give, when applied to code, which is why I prefer this over a boolean value. We could use an enum with the same effect, while also capturing the query param value. Maybe that's the better approach.